pkg/resourcequotas/resourcequotas_test.go

```diff
@@ -0,0 +1,361 @@
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "resource quota 'namespace' cannot be empty",
+			apiClientNil:        false,
+		},
+		{
+			testName:            "test4",
+			testNamespace:       "testNamespace",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "apiClient is nil",
+			apiClientNil:        true,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var runtimeObjects []runtime.Object
+
+		var testSettings *clients.Settings
+
+		testResourceQuota := generateResourceQuota(testCase.testName, testCase.testNamespace)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testResourceQuota)
+		}
+
+		testSettings = clients.GetTestClients(clients.TestClientParams{
+			K8sMockObjects: runtimeObjects,
+		})
+
+		if testCase.apiClientNil {
+			testSettings = nil
+		}
+
+		builderResult, err := Pull(testSettings, testCase.testName, testCase.testNamespace)
+
+		if testCase.expectedError {
+			assert.NotNil(t, err)
+
+			if testCase.expectedErrorText != "" {
+				assert.Equal(t, testCase.expectedErrorText, err.Error())
+			}
+		} else {
+			assert.Nil(t, err)
+			assert.Equal(t, testResourceQuota.Name, builderResult.Object.Name)
+			assert.Equal(t, testResourceQuota.Namespace, builderResult.Object.Namespace)
+		}
+	}
+}
+
+func TestResourceQuotaWithQuotaSpec(t *testing.T) {
+	testCases := []struct {
+		builderNil    bool
+		definitionNil bool
+		expectedError bool
+		errorMessage  string
+	}{
+		{ // Test Case 1 - Happy path, no nils
+			builderNil:    false,
+			definitionNil: false,
+			expectedError: false,
+		},
+		{ // Test Case 2 - Builder is nil
+			builderNil:    true,
+			definitionNil: false,
+			expectedError: true,
+		},
+		{ // Test Case 3 - Definition is nil
+			builderNil:    false,
+			definitionNil: true,
+			expectedError: true,
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := &Builder{
+			apiClient: nil,
+			Definition: &corev1.ResourceQuota{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "testRQ",
+					Namespace: "testNamespace",
+				},
+			},
+		}
+
+		if testCase.builderNil {
+			testBuilder = nil
+		}
+
+		if testCase.definitionNil {
+			testBuilder.Definition = nil
+		}
+
+		result := testBuilder.WithQuotaSpec(corev1.ResourceQuotaSpec{
+			Hard: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("1"),
+				corev1.ResourceMemory: resource.MustParse("1Gi"),
+			},
+		})
+
+		if !testCase.expectedError {
+			assert.NotNil(t, result)
+			assert.NotNil(t, testBuilder.Definition)
+			assert.Equal(t, resource.MustParse("1"), testBuilder.Definition.Spec.Hard[corev1.ResourceCPU])
+			assert.Equal(t, resource.MustParse("1Gi"), testBuilder.Definition.Spec.Hard[corev1.ResourceMemory])
+		} else {
+			assert.Nil(t, result)
+		}
+	}
+}
+
+func TestResourceQuotaCreate(t *testing.T) {
+	generateResourceQuota := func(name, namespace string) *corev1.ResourceQuota {
```

Comment:
Can be moved into a function instead of a closure since it's used in multiple different functions

Reply:
Done.